tests/integrationv2/utils.py

```diff
@@ -47,15 +48,33 @@ def invalid_test_parameters(*args, **kwargs):
     cipher = kwargs.get('cipher')
     curve = kwargs.get('curve')
 
+    if protocol is Protocols.TLS13 and get_flag(S2N_OPENSSL_VERSION) == "openssl-1.0.2-fips":
+        return True
+
     if cipher is not None:
         # If the selected protocol doesn't allow the cipher, don't test
-        if protocol is not None and cipher.min_version > protocol:
-            return True
-
-        # NOTE: We don't detect the version of OpenSSL at the moment,
-        # so we will deselect these tests.
-        if provider is not None and provider is OpenSSL and cipher.openssl1_1_1 is False:
-            return True
+        if protocol is not None:
+            if cipher.min_version > protocol:
+                return True
+            if protocol is Protocols.TLS13 and cipher.min_version < protocol:
+                return True
+
+        if provider is OpenSSL:
+            if get_flag(S2N_OPENSSL_VERSION) == "openssl-1.1.1" and cipher.openssl1_1_1 is False:
+                # If the cipher is not supported in OpenSSL 1.1.1, deselect the test
+                return True
+            if get_flag(S2N_OPENSSL_VERSION) == "openssl-1.0.2-fips":
```

Comment:
Maybe check S2N_OPENSSL_VERSION for the word "fips" rather than looking for a specific version?

Reply:
For the parity goal, this is sufficient. This tells us exactly what we are testing against.